community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -404,6 +406,17 @@ def sync_slurm():
     for status, nodes in node_statuses.items():
         do_node_update(status, nodes)
 
+def future_reservation_sync(nodeset:object, nodename:str, state:object)-> None:
+    if future_reservation := lookup().future_reservation(nodeset.future_reservation, nodeset.zone_policy_allow):
+        currentTime = datetime.utcnow()
+        if future_reservation.nodesAreActive is False and future_reservation.startTime>currentTime and state.base != "DOWN":
+            run(f"{lookup().scontrol} update nodename={nodename} state=down reason='Waiting for reservation to start at {future_reservation.startTime}'")
+
+        if future_reservation.nodesAreActive is False and future_reservation.endTime<currentTime and state.base != "DOWN":
+            run(f"{lookup().scontrol} update nodename={nodename} state=down reason='Reservation is over'")
+
+        if future_reservation.nodesAreActive and state.base == "DOWN":
+            run(f"{lookup().scontrol} update nodename={nodename} state=idle")
```

Comment:
Transition from `down -> idle` should be done with `state=resume`. This is tribal knowledge, and I don't have good answer "why" ) We can dig into it.

Reply:
It appears to work with idle but it also appears to have similar behaviour to `state=resume`.

->Will usage of state=resume cause nodes to be provisioned (go through resume.py) (unwanted)
In the sense that `srun` causes resume script to run? If so yes (for both idle and resume) but isn't this what we'd like? Or are you asking if `state=idle` automatically triggers `resume.py` which is not the case (nor for `state=resume`). Also wondering on your thoughts about `state=down` vs `state=power_down` considering last week's discussion. Functionality wise it seems like the use of `state=power_down` always calls `suspend.py`.